File: test code/Day 05/src/test/Test7.java
package test;

/*
生成10个1~100之间的数存入数组。
1）求出所有数据的和
2）求所有数据的平均值
3）统计有多少个数据比平均值小
 */

import java.util.Random;

public class Test7 {
    public static void main(String[] args) {
        // 创建Random对象
        Random r = new Random();
        // 生成1~30个1~100之间的数存入数组
        int x = r.nextInt(30) + 1;
        int[] arr = new int[x];
        for (int i = 0; i < arr.length; i++) {
            int num = r.nextInt(100) + 1;
            arr[i] = num;
            // 遍历数组
            System.out.print(arr[i] + "  ");
        }
        // 求出所有数据的和
        int sum = 0;
        for (int i = 1; i < arr.length; i++) {
            sum += arr[i];
        }
        System.out.println();
        System.out.println("所有数据的和为：" + sum);
        // 求所有数据的平均值
        int avg = sum / arr.length;
        System.out.println("所有数据的平均值为：" + avg);
        // 统计有多少个数据比平均值小
        int count = 0;
        for (int i = 0; i < arr.length; i++) {
            if (arr[i] < sum) {
                count++;
            }
        }
        System.out.println("有 " +  count + " 个数据比平均值小");
    }
}

File: Python/test code/流程控制语句/test7.py
import random
num = random.randint(1, 10)
print("答案", num)
guess = int(input("请输入你猜的数字："))

if guess == num:
    print("恭喜你，猜对了！")
else:
    if guess > num:
        print("大了")
    else:
        print("小了")
        
    guess = int(input("再猜一次："))
    if guess == num:
        print("恭喜您猜对了")
    else:
        if guess > num:
            print("大了")
        else:
            print("小了")
        
        guess = int(input("再猜一次："))
        if guess == num:
            print("恭喜您猜对了")
        else:
            if guess > num:
                print("大了，答案是", num)
            else:
                print("小了答案是", num)